images/win/scripts/Installers/Install-Chrome.ps1

```diff
@@ -2,25 +2,116 @@
+        }
+        else {
+            Write-Debug "Try to stop service [$ServiceName]";
+            try {
+                Stop-Service -Name $ServiceName -Force;
+                $Service.WaitForStatus("Stopped", "00:01:00");
+                Write-Debug "Service [$ServiceName] has been stoppet successfuly";
+            }
+            catch {
+                Write-Error "[!] Failed to stop service [$ServiceName] with error:"
+                $_ | Out-String | Write-Error;
+            }
+        }
+    }
+    End {
+        Write-Debug "Function [Stop-SvcWithErrHnadlig] is stopped";
+    }
+}
 
-$temp_install_dir = 'C:\Windows\Installer'
-New-Item -Path $temp_install_dir -ItemType Directory -Force
+function Set-SvcWithErrHandling 
+{
+    param (
+        [Parameter(Mandatory, ValueFromPipeLine = $true)] [string] $ServiceName,
+        [Parameter(Mandatory)] [hashtable] $Arguments
+    )
+    Begin { 
+        Write-Debug "Function [Set-SvcWithErrHnadlig] is started"; 
+    }
+    Process {
+        $Service = Get-Service $ServiceName -ErrorAction SilentlyContinue
+        if(-not $Service) {
+            Write-Warning "[!] Service [$ServiceName] is not found";
+        }
+        try {
+           Set-Service $ServiceName @Arguments;
+        }
+        catch {
+            Write-Error "[!] Failed to set service [$ServiceName] arguments with error:"
+            $_ | Out-String | Write-Error;
+        }
+    }
+    End {
+        Write-Debug "Function [Stop-SvcWithErrHnadlig] is stopped";
+    }
+}
 
-Install-MSI -MsiUrl "https://seleniumwebdrivers.blob.core.windows.net/knownchromeversion/googlechromestandaloneenterprise64.msi" -MsiName "googlechromestandaloneenterprise64.msi"
+function New-ItemWithErrHandling {
+    param (
+        [Parameter(Mandatory)] [hashtable] $Arguments
+    )
+    Write-Debug "Creation of [$($Arguments.Name)] item";
+    try {
+        New-ItemProperty @Arguments;
+    }
+    catch {
+        Write-Warning "[!] Failed to create [$($Arguments.Name)] registry parameter";
+    }
+}
 
-New-NetFirewallRule -DisplayName "BlockGoogleUpdate" -Direction Outbound -Action Block -Program "C:\Program Files (x86)\Google\Update\GoogleUpdate.exe"
+Import-Module -Name ImageHelpers -Force;
 
-Stop-Service -Name gupdate -Force
-Set-Service -Name gupdate -StartupType "Disabled"
-Stop-Service -Name gupdatem -Force
-Set-Service -Name gupdatem -StartupType "Disabled"
+$ChromeInstallerFile = "chrome_installer.exe";
+$ChromeInstallerUri = "https://dl.google.com/chrome/install/375.126/chrome_installer.exe";
+Install-Exe -Url $ChromeInstallerUri -Name $ChromeInstallerFile -ArgumentList ("/silent", "/install")
+
+Write-Debug "Adding the firewall rule for Google update blocking";
+New-NetFirewallRule -DisplayName "BlockGoogleUpdate" -Direction Outbound -Action Block -Program "C:\Program Files (x86)\Google\Update\GoogleUpdate.exe";
+
+('gupdate','gupdatem') | Stop-SvcWithErrHandling;
```

Comment:
Nitpick: ('gupdate','gupdatem') set to a variable and other changes only at one place.

Reply:
('gupdate','gupdatem') array has been encapsulated to variable